packages/bsky/src/api/app/bsky/graph/getSuggestedFollowsByActor.ts

```diff
@@ -0,0 +1,132 @@
+import { sql } from 'kysely';
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+const MAX_RESULTS_LENGTH = 10
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.graph.getSuggestedFollowsByActor({
+    auth: ctx.authVerifier,
+    handler: async ({ auth, params }) => {
+      const { actor } = params
+      const viewer = auth.credentials.did
+
+      const db = ctx.db.getReplica()
+      const actorService = ctx.services.actor(db)
+      const actorDid = await actorService.getActorDid(actor)
+
+      if (!actorDid) {
+        throw new InvalidRequestError('Actor not found')
+      }
+
+      const actorsViewerFollows = db.db
+        .selectFrom('follow')
+        .where('creator', '=', viewer)
+        .select('subjectDid')
+
+      /**
+       * 20 most liked accounts that aren't already followed by the viewer, ARE
+       * the viewer, or are the actor
+       */
+      const mostLikedAccounts = db.db
+        .selectFrom(
+          db.db
+            .selectFrom('like')
+            .where('creator', '=', actorDid)
+            .select(sql`split_part(subject, '/', 3)`.as('subjectDid'))
+            .as('likes')
+        )
+        .select('likes.subjectDid as did')
+        .select(qb => qb.fn.count('likes.subjectDid').as('count'))
+        .where('likes.subjectDid', 'not in', actorsViewerFollows)
+        .where('likes.subjectDid', 'not in', [actorDid, viewer])
```

Comment:
Have to do this kind of thing in a few places, maybe could combine? Just don't want to overfetch, cuz if we have to filter out "actors viewer already follows" later, it could cut down on the # of results substantially (if the viewer follows a lot of accounts in common with the actor).

Doing it in each query gives us as many good results as possible each time 🤔

Reply:
yeah this seems fine 👍 it's a _bit_ wordy but i prefer wordy over unclear